test/e2e/tests/confirmations/signatures/signature-request.spec.ts

```diff
@@ -0,0 +1,269 @@
+import SignTypedData from '../../../page-objects/pages/confirmations/redesign/sign-typed-data-confirmation';
+
+import {
+  withFixtures,
+  openDapp,
+  unlockWallet,
+  WINDOW_TITLES,
+} from '../../../helpers';
+import FixtureBuilder from '../../../fixture-builder';
+import TestDapp from '../../../page-objects/pages/test-dapp';
+import { Driver } from '../../../webdriver/driver';
+
+const signatureRequestType = {
+  signTypedData: 'Sign Typed Data',
+  signTypedDataV3: 'Sign Typed Data V3',
+  signTypedDataV4: 'Sign Typed Data V4',
+};
+
+const testData = [
+  {
+    type: signatureRequestType.signTypedData,
+  },
+  {
+    type: signatureRequestType.signTypedDataV3,
+  },
+  {
+    type: signatureRequestType.signTypedDataV4,
+  },
+];
+
+describe('Sign Typed Data Signature Request', function () {
+  testData.forEach((data) => {
+    it(`can queue multiple Signature Requests of ${data.type} and confirm`, async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .build(),
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver, localNodes }) => {
+          const confirmation = new SignTypedData(driver);
+          const addresses = await localNodes[0].getAccounts();
+          const publicAddress = addresses[0].toLowerCase();
+          await unlockWallet(driver);
+
+          await openDapp(driver);
+
+          // creates multiple sign typed data signature requests
+          await triggerSignatureRequest(driver, data.type);
+
+          await driver.waitUntilXWindowHandles(3);
+          const windowHandles = await driver.getAllWindowHandles();
+          // switches to Dapp
+          await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+          // creates second sign typed data signature request
+          await triggerSignatureRequest(driver, data.type);
+
+          await driver.switchToWindowWithTitle(
+            WINDOW_TITLES.Dialog,
+            windowHandles,
+          );
```

Comment:
could we use this instead:

```suggestion
          await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
```